### PR TITLE
1064_derive_param_merged_exist_flag: Implement `derive_param_merged_exist_flag()`

### LINKED ISSUE #1064: Feature Request: Implement `derive_param_merged_exist_flag()`
### Feature Idea

```
derive_var_merged_exist_flag(
  dataset,
  dataset_adsl,
  dataset_add,
  by_vars,
  condition,
  true_value = "Y",
  false_value = NA_character_,
  missing_value = NA_character_,
  filter_add = NULL,
  set_values_to
)
```
adds a new parameter to the input dataset. An observation for each subject in `dataset_adsl` is added. `AVALC` is set to `true_value` if the condition is fulfilled for any observation in `dataset_add`. Otherwise it is set to `false_value`. For subject without an observation in `dataset_add` it is set to `missing_value`. `AVAL` is set to `1`, `0`, or `NA`.

### Relevant Input

What should the input look like? REMINDER: no patient level data or company sensitive information should be shared via this open public issue

### Relevant Output

What should the output look like? REMINDER: no patient level data or company sensitive information should be shared via this open public issue

### Reproducible Example/Pseudo Code

```
derive_param_merged_exist_flag(
  dataset = adrs,
  dataset_adsl = adsl,
  dataset_add = tu,
  condition = TUTESTCD == 'TUMIDENT' & TUEVAL == 'INVESTIGATOR' & TUSTRESC = 'TARGET' & convert_dtc_to_dt(TUDTC) < TRTSDT,
  false_value = "N",
  missing_value = "N"
  set_values_to = vars(
    PARAMCD = "MDIS",
    PARAM = "Measurable Disease at Baseline by Investigator"
  )
)
```

### PR DESCRIPTION
**Pull request by bundfussr**
Closes #1064 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_merged_exist_flag.R [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  condition <- assert_filter_cond(enquo(condition))
+  assert_character_scalar(true_value)
+  assert_character_scalar(false_value)
+  assert_character_scalar(missing_value)
+  filter_add <- assert_filter_cond(enquo(filter_add), optional = TRUE)
+  assert_function(aval_fun)
+  assert_vars(subject_keys)
+  assert_data_frame(dataset_adsl, required_vars = subject_keys)
+  assert_data_frame(dataset_add, required_vars = subject_keys)
+  assert_varval_list(set_values_to)
```
Comment (kabis-ops) [inline R/derive_param_merged_exist_flag.R:184]: We should check PARAM or PARAMCD to be in 'set_values_to'

File: R/derive_param_merged_exist_flag.R [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                                           set_values_to) {
+  # Check input parameters
+  condition <- assert_filter_cond(enquo(condition))
+  assert_character_scalar(true_value)
+  assert_character_scalar(false_value)
+  assert_character_scalar(missing_value)
+  filter_add <- assert_filter_cond(enquo(filter_add), optional = TRUE)
+  assert_function(aval_fun)
+  assert_vars(subject_keys)
+  assert_data_frame(dataset_adsl, required_vars = subject_keys)
```
Comment (kabis-ops) [inline R/derive_param_merged_exist_flag.R:182]: assert_data_frame to be added for `dataset` too

File: tests/testthat/test-derive_param_merged_exist_flag.R [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  "2",      "PD",   "OVRF",
+  "2",      "PD",   "OVRF"
+) %>%
+  mutate(
+    STUDYID = "XX1234",
+    ANL01FL = "Y"
+  )
+
+# derive_param_merged_exist_flag ----
+## derive_param_merge_exist_flag Test 1: derive parameter indicating PD ----
```
Comment (kabis-ops) [inline tests/testthat/test-derive_param_merged_exist_flag.R:32]: Maybe this is ok for testing: but once again i think the PD param needs a date together with AVALC

File: R/derive_param_merged_exist_flag.R [hunk 4]
```diff
@@ -0,0 +1,237 @@
+#' Add an Existence Flag Parameter
+#'
+#' Add a new parameter indicating that a certain event exists in a dataset.
+#' `AVALC` and `AVAL` indicate if an event occurred. For example, the function
+#' can derive a parameter indicating if there is measureable disease at
+#' baseline.
+#'
+#' @param dataset Input dataset
```
Comment (kabis-ops) [inline R/derive_param_merged_exist_flag.R:8]: subject_keys should also be in this datsets

**Review by kabis-ops — COMMENTED** [review]
H @bundfussr , I have added some comments. Please have a look. Thanks

File: R/derive_param_merged_exist_flag.R [hunk 5]
```diff
@@ -0,0 +1,237 @@
+#' Add an Existence Flag Parameter
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:1]: The "merged" comes from `deriive_var_merged_exist_flag()` which is the basic function called to derive the parameter. However, I agree that `derive_parm_exist_flag()` is better. I have renamed it.

File: R/derive_param_merged_exist_flag.R [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+#'
+#' @param filter_add Filter for additional data
+#'
+#'   Only observations fulfilling the specified condition are taken into account
+#'   for flagging. If the parameter is not specified, all observations are
+#'   considered.
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @param condition Event condition
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:66]: Yes, removed.

File: R/derive_param_merged_exist_flag.R [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+#'
+#'   For subjects with at least one observation in `dataset_add` where the condition is fulfilled
+#'   `AVALC` is set to `true_value` and `AVAL` to `1`.
+#'
+#'   For subjects with observations in `dataset_add` but none of them is
+#'   fulfilling the condition `AVALC` is set to `false_value` and `AVAL` to `0`.
+#'
+#'   For all other subjects, i.e., those without an observation in `dataset_add`
+#'   `AVALC` is set to `missing_value`, `AVAL` to `0`, and `ADT` to `NA`.
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:75]: Yes

File: R/derive_param_merged_exist_flag.R [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+#'   the true value (`true_value`) for the new observations. For all subjects
+#'   where it evaluates as `FALSE` or `NA` for all observations `AVALC` is set
+#'   to the false value (`false_value`). `AVALC` is set to the missing value
+#'   (`missing_value`) for subjects not present in the additional dataset.
+#'
+#' @param true_value True value
+#'
+#'   For the new observations `AVALC` is set to the specified value for all
+#'   subjects with at least one observations in the additional dataset
+#'   (`dataset_add`) fulfilling the event condition (`condition`).
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:33]: Updated

File: R/derive_param_merged_exist_flag.R [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+#'   For subjects with at least one observation in `dataset_add` where the condition is fulfilled
+#'   `AVALC` is set to `true_value` and `AVAL` to `1`.
+#'
+#'   For subjects with observations in `dataset_add` but none of them is
+#'   fulfilling the condition `AVALC` is set to `false_value` and `AVAL` to `0`.
+#'
+#'   For all other subjects, i.e., those without an observation in `dataset_add`
+#'   `AVALC` is set to `missing_value`, `AVAL` to `0`, and `ADT` to `NA`.
+#'
+#' @param aval_fun Function to map `AVALC` values to `AVAL`
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:77]: Updated

File: R/derive_param_merged_exist_flag.R [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+#'   1. The `AVAL` variable is added and set to `aval_fun(AVALC)`.
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @return The input dataset with a new parameter indicating if and when an
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:117]: No, `ADT` is not calculated. This is a copy and paste error. Fixed now.

File: R/derive_param_merged_exist_flag.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:184]: Added

File: R/derive_param_merged_exist_flag.R (hunk 2 above)
Comment (bundfussr) [reply R/derive_param_merged_exist_flag.R:182]: Added

File: NEWS.md [hunk 11]
```diff
@@ -37,6 +37,9 @@ from a BDS dataset, e.g. ADLB. (#943)
 - `derive_param_wbc_abs()` - Adds a parameter for lab differentials converted to absolute values. 
 (#941)
 
+- New function `derive_param_merged_exist_flag()`: Add a new parameter
```
Comment (kabis-ops) [inline NEWS.md:40]: Please update the name of the function

Comment (bundfussr) [reply NEWS.md]: Thanks, updated.

File: R/derive_param_exist_flag.R [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @return The input dataset with a new parameter indicating if an event
+#'   occurred
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_param_exist_flag.R]: Done

**Review by kabis-ops — APPROVED** [review]

**Comment by bundfussr** [comment]
@kabis-ops , I have merge in the latest changes from `devel`. Could you approve again to start the check_templates check?

**Review by kabis-ops — APPROVED** [review]

**Pull request merged**